dbt_semantic_interfaces/validations/labels.py

```diff
@@ -41,3 +42,103 @@ def validate_manifest(semantic_manifest: SemanticManifestT) -> Sequence[Validati
             issues += MetricLabelsRule._check_metric(metric=metric, existing_labels=labels_to_metrics)
 
         return issues
+
+
+class SemanticModelLabelsRule(SemanticManifestValidationRule[SemanticManifestT], Generic[SemanticManifestT]):
+    """Checks that the labels are unique across semantic models."""
+
+    @staticmethod
+    @validate_safely("checking that a semantic model has a unique label")
+    def _check_semantic_model(
+        semantic_model: SemanticModel, existing_labels: Dict[str, str]
+    ) -> Sequence[ValidationIssue]:  # noqa: D
+        if semantic_model.label in existing_labels:
+            return (
+                ValidationError(
+                    context=FileContext.from_metadata(semantic_model.metadata),
+                    message=f"Can't use label `{semantic_model.label}` for  semantic model `{semantic_model.name}` "
+                    f"as it's already used for semantic model `{existing_labels[semantic_model.label]}`",
+                ),
+            )
+        elif semantic_model.label is not None:
+            existing_labels[semantic_model.label] = semantic_model.name
+
+        return ()
+
+    @staticmethod
+    @validate_safely("checking that a semantic model's dimension labels are unique within itself")
+    def _check_semantic_model_dimensions(semantic_model: SemanticModel) -> Sequence[ValidationIssue]:
+        issues: List[ValidationIssue] = []
+        label_counts: DefaultDict[str, int] = defaultdict(lambda: 0)
+        for dimension in semantic_model.dimensions:
+            if dimension.label is not None:
+                label_counts[dimension.label] = label_counts[dimension.label] + 1
+
+        for label, count in label_counts.items():
+            if count > 1:
+                issues.append(
+                    ValidationError(
+                        context=FileContext.from_metadata(semantic_model.metadata),
+                        message=f"Dimension labels must be unique within a semantic model. The label `{label}` was "
```

Comment:
Do we want them to be globally unique for dimensions? I can see people setting `user__country` and `sales_rep__country` to both be `Country`, which isn't great.

Reply:
For those following along, we discussed this in a call. The decision for now was to not require labels for dimensions or measures to be globally unique. If people end up consistently running into disambiguation problems, we'll look into being stricter